CMakeLists.txt

```diff
@@ -90,6 +90,63 @@ list(APPEND rcutils_sources
   include/rcutils/logging_macros.h)
 include_directories("${CMAKE_CURRENT_BINARY_DIR}/include")
 
+###############################################################################
+# Create version file in build/<package>/include/<package> folder which is already in the include list
+set(TMP_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/include/${PROJECT_NAME})
+set(RCUTILS_VERSION_FILE_NAME ${TMP_INCLUDE_DIR}/version.h)
+set(NEED_TO_CREATE_VERSION_FILE false)
+
+# retrieve verson information from <package>.xml file
+# call ament_package_xml() if it has not been called before
+if(NOT _AMENT_PACKAGE_NAME)
```

Comment:
Hm, I am not happy that the code need to check this variable which it shouldn't need to know about. Maybe we should update the `ament_package_xml()` function to perform that check internally and when being called repeatedly not error out but just be a no-op?

Reply:
@dirk-thomas,

I assume it's a different issue outside of the scope of this one